src/vs/code/electron-main/windows.ts

```diff
@@ -1144,13 +1144,20 @@ export class WindowsManager implements IWindowsMainService {
 		// Update in settings
 		const menuBarHidden = this.storageService.getItem(VSCodeWindow.menuBarHiddenKey, false);
 		const newMenuBarHidden = !menuBarHidden;
+
+		const windowConfig = this.configurationService.getConfiguration<IWindowSettings>('window');
```

Comment:
It is true that on the main side we do not have access to the default values because they get defined on the browser side. Maybe we should turn this setting around and make it so that the default of a false value is OK for us. That avoids having to fill in a default on our end (instead of having `true` as default, change the setting so that `false` as default is good).